docs/tutorial/typesfuns.rst

```diff
@@ -661,6 +661,93 @@ without any explicit use of ``Force`` or ``Delay``:
     ifThenElse True  t e = t;
     ifThenElse False t e = e;
 
+Codata Types
+============
+
+Codata types are like regular data types, except that they allow for us to
+define infinite data structures.  More precisely, for a type ``T``, each of its
+constructor arguments of type ``T`` are transformed into a coinductive parameter
+``Inf T``. This makes each of the ``T`` arguments lazy, and allows infinite data
+structures of type ``T`` to be built. One example of a codata type is Stream,
+which is defined as follows.
+
+.. code-block:: idris
+
+    codata Stream : Type -> Type where
+      (::) : (e : a) -> Stream a -> Stream a
+
+This gets translated into the following by the compiler.
+
+.. code-block:: idris
+
+    codata Stream : Type -> Type where
```

Comment:
I think you can just say it translates into `data Stream ...`? I mean, you could write the impl of Stream with `data` + `Inf` instead of `codata` and the end result would be the same?